frontend/src/metabase/lib/settings.js

```diff
@@ -88,12 +88,7 @@ class Settings {
   // Right now, all Metabase Cloud hosted instances run on *.metabaseapp.com
   // We plan on changing this to look at an envvar in the future instead.
   isHosted() {
-    return (
-      this.get("site-url") &&
-      this.get("site-url")
-        .toLowerCase()
-        .includes("metabaseapp.com")
-    );
+    return /.+\.metabaseapp.com$/i.test(this.get("site-url"));
```

Comment:
As I said in [this comment](https://github.com/metabase/metabase/pull/14511/files#r564144899), we don't have to use one-line solution. I don't have a strong preference towards one or the other.

```javascript
isHosted() {
  const regex = new RegExp(/.+\.metabaseapp.com$/i)
  return regex.test(this.get("site-url"));
}
```

Reply:
I think the `$` makes it too restrictive. Site URLs can include a path or a port e.g. `foo.metabaseapp.com:80/bar`. Though, I'm not sure why someone would add that for a hosted instance, so probably not a big deal.

The "right" solution is to use `URL` to parse the string for us. The annoying case to handle there is that `new URL`  throws an error if the string isn't a valid URL.

Reply:
That CodeQL warning was complaining exactly about `metabaseapp.com` being positioned randomly in the URL. That's why I tried to make sure it ends with it.
But I didn't know that URL can include a path or a port.

If this can lead to unpredictable behavior, we can close it and live with that warning, AFAIC.

Reply:
...or maybe use something like:

`/.+\.metabaseapp.com(:\d+)?(\/.+)?$/i`

Reply:
Ah I see. Probably fine to merge this then. As long as there's not a trailing `/` by default, I don't think people can move their hosted instance to a subpath and adding `:80` would be pointless.

Reply:
This is one more +1 for envvar. Hosted cannot use sub-dir (and basically not other ports either).

Reply:
> ...or maybe use something like: `/.+\.metabaseapp.com(:\d+)?(\/.+)?$/i`

As a general rule, I wouldn't do that. It's often tempting to parse using regex, but the implementation almost always forgets things. Relying on the browser's builtin parser with `new URL()` is much safer, but if we don't care in this case going with the first regex or `endsWith` seems fine (assuming the site URL doesn't end with `/` by default).